web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -15,45 +22,117 @@
+            @ApiParam("Direction of the sort")
+            @RequestParam(defaultValue = "ASC") Direction direction) {
 
-        throw new UnsupportedOperationException();
+        if (projection == Projection.META) {
+            HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.add(HeaderKeyConstants.TOTAL_COUNT, sampleService.getMetaSamplesInStudy(studyId)
+                    .getTotalCount().toString());
+            return new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<List<? extends SampleSummary>>(
+                    sampleService.getAllSamplesInStudy(studyId, projection.name(), pageSize, pageNumber,
+                            sortBy == null ? null : sortBy.name(), direction.name()), HttpStatus.OK);
+        }
     }
 
-    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}", method = RequestMethod.GET)
+    @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get a sample in a study")
-    public ResponseEntity<SampleSummary> getSampleInStudy(@PathVariable String studyId,
-                                                          @PathVariable String sampleId) {
+    public ResponseEntity<Sample> getSampleInStudy(
+            @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
+            @PathVariable String studyId,
+            @ApiParam(required = true, value = "Sample ID e.g. TCGA-OR-A5J2-01")
+            @PathVariable String sampleId) throws SampleNotFoundException {
 
-        throw new UnsupportedOperationException();
+        return new ResponseEntity<>(sampleService.getSampleInStudy(studyId, sampleId), HttpStatus.OK);
     }
 
-    @RequestMapping(value = "/samples/fetch", method = RequestMethod.POST)
-    @ApiOperation("Fetch samples by ID")
-    public ResponseEntity<List<? extends SampleSummary>> fetchSamples(@RequestParam(defaultValue = "SUMMARY") Projection projection,
-                                                                      @RequestBody List<SampleIdentifier> sampleIdentifiers) {
+    @RequestMapping(value = "/studies/{studyId}/patients/{patientId}/samples", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all samples of a patient in a study")
+    public ResponseEntity<List<? extends SampleSummary>> getAllSamplesOfPatientInStudy(
+            @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
+            @PathVariable String studyId,
+            @ApiParam(required = true, value = "Patient ID e.g. TCGA-OR-A5J2")
+            @PathVariable String patientId,
+            @ApiParam("Level of detail of the response")
+            @RequestParam(defaultValue = "SUMMARY") Projection projection,
+            @ApiParam("Page size of the result list")
```

Comment:
For pagesize arguments, it would be good to communicate the absolute maximum value [unless this is already covered in a general "using the API" document.

I'm thinking that we may want different absolute upper limits based on the expected side of the data records, so the limit may be endpoint-specific. (I see here there is one max -- PagingConstants.MAX_PAGE_SIZE)